src/shared/sleep-config.c

```diff
@@ -271,5 +349,19 @@ int can_sleep(const char *verb) {
         if (!can_sleep_state(states) || !can_sleep_disk(modes))
                 return false;
 
+        /* We keep an optional white and black list (by product) to
+         * control if we want to explicitly support sleep operations. */
+        r = parse_sleep_products_config(verb, &blacklist, &whitelist);
+        if (r < 0)
+                return false;
+
+        if (is_product_listed(blacklist))
+                return false;
+
+        /* We don't support sleep operations on desktop computers
+           unless the product is explcitly white listed. */
+        if (!access("/sys/class/power_supply/BAT0", F_OK) == 0 && !is_product_listed(whitelist))
```

Comment:
Not sure what you mean here... are you proposing to keep this logic but replace the "is desktop/laptop" check with something using hostnamed's chassis property?

I ask this because the whitelisting does not make much sense if we remove that logic of not supporting desktops by default, as there's nothing to whitelist, as we can't simply return true here because the final check would still be done after this lines:
```
    return streq(verb, "suspend") || enough_memory_for_hibernation();
```

So, I'm fine not adding any extra logic to avoid supporting desktop computers by default, but if we don't do that, then there's nothing to whitelist at this level, and only blacklisting makes sense IMHO.